### PR TITLE
Governance feature cases

### PR DESCRIPTION
**Pull request by jtsang586**
This PR contains:
Updated feature file for Cypress Governance tests 
Steps for checking each proposal including fields are displayed
Mocks for proposals - merged from `fix/1104-proposals-mocks`
Added test ids for proposals

### CODE REVIEW

File: src/components/graphql-provider/__mocks__/graphql-provider.tsx [hunk 1]
```diff
@@ -382,11 +542,29 @@ export const GraphQlProvider = ({
...
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
+        ...MOCK_PROPOSAL,
```
Comment (dexturr) [inline src/components/graphql-provider/__mocks__/graphql-provider.tsx:564]: Need a better solution than this for serving multiple mocks, I said I'd have a look into it before xmas but didn't have the time. Will look now 👍

File: src/components/graphql-provider/__mocks__/graphql-provider.tsx [hunk 2]
```diff
@@ -382,11 +542,29 @@ export const GraphQlProvider = ({
   return (
     <MockedProvider
       mocks={[
```
Comment (pgriscti) [inline src/components/graphql-provider/__mocks__/graphql-provider.tsx:544]: Is there a way around this?

File: src/components/graphql-provider/__mocks__/graphql-provider.tsx (hunk 1 above)
Comment (dexturr) [reply src/components/graphql-provider/__mocks__/graphql-provider.tsx:564]: Open issue about this:
https://github.com/apollographql/apollo-feature-requests/issues/274

Potential work around:
https://gist.github.com/geraint-37d5/626f5c68d6f943a9c9d4508c4e18e564

Let's all have a discussion about it in our next standup

Comment (Ditmir-Vega) [reply src/components/graphql-provider/__mocks__/graphql-provider.tsx:564]: can this get merged in like this and handled on a seperate PR ?

File: src/routes/governance/test-helpers/generate-proposals.ts [hunk 3]
```diff
@@ -68,3 +60,63 @@ export function generateProposal(
...
+            currentStakeAvailable: faker.datatype
+              .number({
+                min: 1,
+                max: 10000,
+              })
+              .toString(),
+          },
+        },
+        datetime: faker.date.past().toISOString(),
+      } as ProposalFields_votes_no_votes;
```
Comment (dexturr) [inline src/routes/governance/test-helpers/generate-proposals.ts:119]: Redundant type cast
```suggestion
      };
```

Comment (dexturr) [reply src/routes/governance/test-helpers/generate-proposals.ts:119]: Also on line 89

File: src/routes/governance/test-helpers/generate-proposals.ts [hunk 4]
```diff
@@ -3,7 +3,13 @@ import merge from "lodash/merge";
 
 import { ProposalState, VoteValue } from "../../../__generated__/globalTypes";
 import { DeepPartial } from "../../../lib/type-helpers";
-import { ProposalFields } from "../__generated__/ProposalFields";
+import {
+  ProposalFields,
+  ProposalFields_votes_no,
+  ProposalFields_votes_no_votes,
```
Comment (dexturr) [inline src/routes/governance/test-helpers/generate-proposals.ts:9]: Not used, can we please remove
```suggestion
```

File: src/routes/governance/proposals-list.tsx [hunk 5]
```diff
@@ -44,11 +44,14 @@ export const ProposalsList = ({ proposals }: ProposalsListProps) => {
             </p>
           </header>
         </Link>
-        <table className="proposal-table">
+        <table
+          className="proposal-table"
+          data-test-id="governance-proposal-table"
```
Comment (dexturr) [inline src/routes/governance/proposals-list.tsx:49]: can all `data-test-id`s be changed to `data-testid` the reason for this is Jest uses the latter so we can use the ids in both unit and e2e tests if they are in the latter form but not in the former.

File: automation/cypress/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (dexturr) [inline automation/cypress/tsconfig.json:1]: Why needed?

**Review by dexturr — CHANGES_REQUESTED** [review]
Mostly good few minor niggles that should be quick to fix.

File: automation/cypress/tsconfig.json (hunk 6 above)
Comment (jtsang586) [reply automation/cypress/tsconfig.json:1]: Not sure if this is just my machine. But I was getting undefined Cypress commands which is fixed by the tsconfig file found in Cypress docs. Happy to remove this if it's breaking something or if there's an alternative to fixing the problem I was getting.

File: src/components/graphql-provider/__mocks__/graphql-provider.tsx (hunk 2 above)
Comment (jtsang586) [reply src/components/graphql-provider/__mocks__/graphql-provider.tsx:544]: This looks like a temporary solution which will be properly fixed at some point soon.

File: src/routes/governance/proposals-list.tsx (hunk 5 above)
Comment (jtsang586) [reply src/routes/governance/proposals-list.tsx]: Done

File: src/routes/governance/test-helpers/generate-proposals.ts (hunk 3 above)
Comment (jtsang586) [reply src/routes/governance/test-helpers/generate-proposals.ts]: Removed

File: automation/cypress/tsconfig.json (hunk 6 above)
Comment (dexturr) [reply automation/cypress/tsconfig.json:1]: Nope happy as long as there is a reason.

File: src/components/graphql-provider/__mocks__/graphql-provider.tsx (hunk 1 above)
Comment (dexturr) [reply src/components/graphql-provider/__mocks__/graphql-provider.tsx:564]: #1117

**Review by dexturr — APPROVED** [review]

File: automation/cypress/support/step_definitions/home.step.js [hunk 7]
```diff
@@ -41,10 +41,10 @@ Then("the vesting address is shown", () => {
 });
 
 Then("the total supply is shown correctly", () => {
-  cy.get('[data-testid="total-supply"]').should(
-    "have.text",
-    "1,000,000,000.00"
-  );
+  cy.get('[data-testid="total-supply"]').then(($supply) => {
+    const supplyNum = Number($supply.text().replace(/,/g, ""));
+    expect(supplyNum).to.be.above(65000000);
```
Comment (Ditmir-Vega) [inline automation/cypress/support/step_definitions/home.step.js:46]: noice !!!

**Review by Ditmir-Vega — APPROVED** [review]
LGTM

**Review by Ditmir-Vega — APPROVED** [review]

**Comment by Ditmir-Vega** [comment]
resolved conflict

File: automation/cypress/support/step_definitions/home.step.js (hunk 7 above)
Comment (dexturr) [reply automation/cypress/support/step_definitions/home.step.js:47]: This is wrong, there can be fewer tha 65M tokens. As you can see on the live mainnet token [here](https://etherscan.io/token/0xcb84d72e61e383767c4dfeb2d8ff7f4fb89abc6e)

```suggestion
    expect(supplyNum).to.be.above(64999723);
```

**Pull request merged**